controllers/lease_controller.go

```diff
@@ -0,0 +1,456 @@
+	}
+
+	if *storeValues.Provider == "Swift" {
+		env = append(env, getEnvVarFromSecrets("OS_AUTH_URL", storeValues.SecretRef.Name, "authURL"))
+		env = append(env, getEnvVarFromSecrets("OS_DOMAIN_NAME", storeValues.SecretRef.Name, "domainName"))
+		env = append(env, getEnvVarFromSecrets("OS_USERNAME", storeValues.SecretRef.Name, "username"))
+		env = append(env, getEnvVarFromSecrets("OS_PASSWORD", storeValues.SecretRef.Name, "password"))
+		env = append(env, getEnvVarFromSecrets("OS_TENANT_NAME", storeValues.SecretRef.Name, "tenantName"))
+	}
+
+	if *storeValues.Provider == "OSS" {
+		env = append(env, getEnvVarFromSecrets("ALICLOUD_ENDPOINT", storeValues.SecretRef.Name, "storageEndpoint"))
+		env = append(env, getEnvVarFromSecrets("ALICLOUD_ACCESS_KEY_SECRET", storeValues.SecretRef.Name, "accessKeySecret"))
+		env = append(env, getEnvVarFromSecrets("ALICLOUD_ACCESS_KEY_ID", storeValues.SecretRef.Name, "accessKeyID"))
+	}
+
+	if *storeValues.Provider == "ECS" {
+		env = append(env, getEnvVarFromSecrets("ECS_ENDPOINT", storeValues.SecretRef.Name, "endpoint"))
+		env = append(env, getEnvVarFromSecrets("ECS_ACCESS_KEY_ID", storeValues.SecretRef.Name, "accessKeyID"))
+		env = append(env, getEnvVarFromSecrets("ECS_SECRET_ACCESS_KEY", storeValues.SecretRef.Name, "secretAccessKey"))
+	}
+
+	if *storeValues.Provider == "OCS" {
+		env = append(env, getEnvVarFromSecrets("OCS_ENDPOINT", storeValues.SecretRef.Name, "endpoint"))
+		env = append(env, getEnvVarFromSecrets("OCS_ACCESS_KEY_ID", storeValues.SecretRef.Name, "accessKeyID"))
+		env = append(env, getEnvVarFromSecrets("OCS_SECRET_ACCESS_KEY", storeValues.SecretRef.Name, "secretAccessKey"))
+		env = append(env, getEnvVarFromSecrets("OCS_REGION", storeValues.SecretRef.Name, "region"))
+		env = append(env, getEnvVarFromSecrets("OCS_DISABLE_SSL", storeValues.SecretRef.Name, "disableSSL"))
+		env = append(env, getEnvVarFromSecrets("OCS_INSECURE_SKIP_VERIFY", storeValues.SecretRef.Name, "insecureSkipVerify"))
+	}
+
+	return env
+}
+
+func getEnvVarFromValues(name, value string) v1.EnvVar {
+	return v1.EnvVar{
+		Name:  name,
+		Value: value,
+	}
+}
+
+/*func getEnvVarFromFields(name, fieldPath string) v1.EnvVar {
+	return v1.EnvVar{
+		Name: name,
+		ValueFrom: &v1.EnvVarSource{
+			FieldRef: &v1.ObjectFieldSelector{
+				FieldPath: fieldPath,
+			},
+		},
+	}
+}*/
+
+func getEnvVarFromSecrets(name, secretName, secretKey string) v1.EnvVar {
+	return v1.EnvVar{
+		Name: name,
+		ValueFrom: &v1.EnvVarSource{
+			SecretKeyRef: &v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: secretKey,
+			},
+		},
+	}
+}
+
+func getCompactJobCommands(etcd *druidv1alpha1.Etcd) []string {
+	command := []string{"" + "etcdbrctl" + " " + "compact"}
+	command = append(command, "--data-dir=/var/etcd/data")
+	command = append(command, "--snapstore-temp-directory=/var/etcd/data/tmp")
```

Comment:
3 new flags need to be added for backup sidecar to be able to be update snapshot leases as part of compaction
```
--enable-etcd-lease-renewal=true
--full-snapshot-lease-name=<full_snap_name>
--delta-snapshot-lease-name<delta_snap_name>
```